src/api/routes/status.ts

```diff
@@ -3,17 +3,29 @@ import * as fs from 'fs';
 import { DataStore } from '../../datastore/common';
 import { ServerStatusResponse } from '@stacks/stacks-blockchain-api-types';
 import { logger } from '../../helpers';
+import { getChainTipCacheHandler, setChainTipCacheHeaders } from '../controllers/cache-controller';
 
-export function createStatusRouter(_: DataStore): express.Router {
+export function createStatusRouter(db: DataStore): express.Router {
   const router = express.Router();
-
-  const statusHandler = (_: Request, res: any) => {
+  const cacheHandler = getChainTipCacheHandler(db);
+  const statusHandler = async (_: Request, res: any) => {
     try {
       const [branch, commit, tag] = fs.readFileSync('.git-info', 'utf-8').split('\n');
       const response: ServerStatusResponse = {
         server_version: `stacks-blockchain-api ${tag} (${branch}:${commit})`,
         status: 'ready',
       };
+      const chainTip = await db.getUnanchoredChainTip();
+      if (chainTip.found) {
+        response.chain_tip = {
+          block_height: chainTip.result.blockHeight,
+          block_hash: chainTip.result.blockHash,
+          index_block_hash: chainTip.result.indexBlockHash,
+          microblock_hash: chainTip.result.microblockHash,
+          microblock_sequence: chainTip.result.microblockSequence,
+        };
+      }
+      setChainTipCacheHeaders(res);
       res.json(response);
     } catch (error) {
       logger.error(`Unable to read git info`, error);
```

Comment:
This has the potential to show a git info file error if the unrelated db query throws for whatever reason, otherwise LGTM!